### PR TITLE
Add introduction page to Contributing Guide

### PR DESCRIPTION
**Pull request by BD103**
As part of the Contributing Guide working group, this adds the introduction page.

This was initially written on [HackMD](https://hackmd.io/@bevy/rktu0EEuC).

Blocked by #1581.

### CODE REVIEW

File: content/contributing/getting-started/_index.md
Comment (alice-i-cecile) [inline content/contributing/getting-started/_index.md:70]:
```suggestion
2. [`bevy-website`]: This is where the [official website] is hosted, along with the quick start, news, contributing book, live examples and other bits of polished content we want to show our users.
```

File: content/contributing/getting-started/_index.md
Comment (alice-i-cecile) [inline content/contributing/getting-started/_index.md:54]: This should be moved up: welcoming + civil discourse belong together and make a nice capstone.

**Comment by BD103** [comment]
I rebased this on top of #1581, so this is now blocked until that is merged. The introduction and layout changes depend too much on each other.

File: content/contributing/introduction.md
Comment (alice-i-cecile) [inline content/contributing/introduction.md:78]:
```suggestion
The `bevy` repo itself contains many smaller subcrates. Several of them can be used by themselves and almost all of them can be modularly replaced. Feel free to dig around and find one that interests you!
```

File: content/contributing/introduction.md
Comment (TrialDragon) [inline content/contributing/introduction.md:47]:
```suggestion
- **Thoughtful public interfaces over maximal configurability**: Symbols and APIs should be private by default. Every public API should be thoughtfully and consistently designed. Don't expose unnecessary internal implementation details. Don't allow users to "shoot themselves in the foot". Favor one "happy path" API over multiple APIs for different use cases.
```
Uppercase the acronym API.

**Review by alice-i-cecile — APPROVED** [review]

File: content/contributing/introduction.md
Comment (TrialDragon) [inline content/contributing/introduction.md:52]:
```suggestion
- **Civil discourse**: We need to collectively discuss ideas, and the best ideas *should* win, but conversations need to remain respectful at all times. Remember that we're all in this together and always follow our [Code of Conduct].
```
Nit-pick, non-blocking: This just seems to read better this way to me.

**Review by TrialDragon — APPROVED** [review]
LGTM.

**Pull request merged**
